File: 697-degree-of-an-array/degree-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findShortestSubArray = function(nums) {
    const count = {}
    for(let i of nums){
        if(i in count){
            count[i]+=1
        }else{
            count[i]=1
        }
    }
    let m = Math.max(...Object.values(count))
    let maax = []
    for(let i in count){
        if (count[i] == m){
            maax.push(Number(i))
        }
    }
    let ans = Infinity
    for(let i of maax){
        a = nums.indexOf(i)
        b = nums.lastIndexOf(i)
        c = Math.abs(a-b)+1
        ans = Math.min(ans,c)
    }
    
    return ans
};

File: 697-degree-of-an-array/degree-of-an-array.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        c = Counter(nums)
        m = max(c.values())
        arr = []
        for i in c:
            if c[i] == m:
                arr.append(i)
        print(arr)
        ans = len(nums)
        for i in arr:
            a = nums.index(i)
            b = len(nums)-1-nums[::-1].index(i)
            c = abs(a-b)+1
            ans = min(ans,c)
        return ans